tests/acceptance/features/apiAntivirus/antivirus.feature

```diff
@@ -498,3 +498,21 @@ Feature: antivirus
       | Virus found in readme.md. Upload not possible. Virus: Win.Test.EICAR_HDB-1 |
     And for user "Brian" the content of the file ".space/readme.md" of the space "new-space" should be "Here you can add a description for this Space."
     And for user "Alice" the content of the file ".space/readme.md" of the space "new-space" should be "Here you can add a description for this Space."
+
+
+  Scenario: try to overwrite a file with virus content in project space share
+    Given using spaces DAV path
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Brian" has been created with default attributes and without skeleton files
+    And user "Alice" has created a space "new-space" with the default quota using the GraphApi
+    And user "Alice" has shared a space "new-space" with settings:
+      | shareWith | Brian  |
+      | role      | editor |
+    And user "Alice" has uploaded a file inside space "new-space" with content "hello world" to "text.txt"
+    When user "Brian" uploads a file "filesForUpload/filesWithVirus/eicar.com" to "text.txt" in space "new-space" using the WebDAV API
+    Then the HTTP status code should be "204"
```

Comment:
I highly doubt HTTP status code 204 is the correct one. 
In my view, If a server returns a 204 No Content status code, it means that the server successfully processed the request and there is no content to send back as a response.
when it comes to file overwriting, the use of a 204 status code might not be standard. Instead, it would be more appropriate for the server to return a status code such as 403 Forbidden (if the user lacks permission to overwrite the file) or 409 Conflict (if there is a conflict preventing the file from being overwritten, such as a file lock or version mismatch).

Reply:
That is true and I think the server does that in the cases where the permission is not enough or the file is locked but in this case (scanning file for virus), scanning of the file is done as a post-processing step https://doc.owncloud.com/ocis/next/deployment/services/s-list/antivirus.html#post-processing

When the antivirus service becomes an on-demand service the expectation shouldn't be `204` like you mentioned

Reply:
It is an async upload thing. :+1: